libs/vkd3d/command.c

```diff
@@ -3263,7 +3266,8 @@ static bool d3d12_command_list_gather_pending_queries(struct d3d12_command_list
     for (i = 0; i < dispatch_count; i++)
     {
         struct dispatch_entry *d = &dispatches[i];
-        memset(query_map, 0, sizeof(*query_map) * query_map_size);
```

Comment:
We could split by start index too and offset I guess, that way query_map_size is the max size of the set of dispatches rather than dispatch + dispatch + dispatch